WordPress/src/main/java/org/wordpress/android/ui/stories/StoryComposerViewModel.kt

```diff
@@ -14,20 +22,41 @@ import org.wordpress.android.ui.notifications.SystemNotificationsTracker
 import org.wordpress.android.ui.posts.EditPostRepository
 import org.wordpress.android.ui.posts.PostEditorAnalyticsSession
 import org.wordpress.android.ui.posts.PostEditorAnalyticsSession.Outcome.CANCEL
+import org.wordpress.android.ui.posts.SavePostToDbUseCase
+import org.wordpress.android.ui.stories.usecase.SetUntitledStoryTitleIfTitleEmptyUseCase
 import org.wordpress.android.util.analytics.AnalyticsUtilsWrapper
+import org.wordpress.android.util.helpers.MediaFile
+import org.wordpress.android.viewmodel.Event
 import javax.inject.Inject
 
 class StoryComposerViewModel @Inject constructor(
     private val systemNotificationsTracker: SystemNotificationsTracker,
     private val saveInitialPostUseCase: SaveInitialPostUseCase,
+    private val savePostToDbUseCase: SavePostToDbUseCase,
+    private val setUntitledStoryTitleIfTitleEmptyUseCase: SetUntitledStoryTitleIfTitleEmptyUseCase,
     private val analyticsUtilsWrapper: AnalyticsUtilsWrapper,
     private val dispatcher: Dispatcher
-) :
-        ViewModel() {
+) : ViewModel(), LifecycleOwner {
+    private val lifecycleRegistry = LifecycleRegistry(this)
```

Comment:
Could you please elaborate why does the VM need to be LifeCycleOwner?

Reply:
Good question. More details here #12501 (comment)